src/Compilers/CSharp/Test/Semantic/Semantics/DelegateTypeTests.cs

```diff
@@ -2441,6 +2678,41 @@ static class E
             CompileAndVerify(source, parseOptions: TestOptions.Regular10, expectedOutput: @"C.M");
         }
 
+        [WorkItem(4674, "https://github.com/dotnet/csharplang/issues/4674")]
+        [Fact]
+        public void OverloadResolution_43()
+        {
+            var source =
+@"using System;
+using System.Linq.Expressions;
+class Program
+{
+    static int F() => 0;
+    static void Main()
+    {
+        var c = new C();
+        c.M(F);
+    }
+}
+class C
+{
+    public void M(Expression e) { Console.WriteLine(""C.M""); }
+}
+static class E
+{
+    public static void M(this object o, Func<int> a) { Console.WriteLine(""E.M""); }
+}";
+
+            CompileAndVerify(source, parseOptions: TestOptions.Regular9, expectedOutput: @"E.M");
+
+            // Breaking change from C#9 which binds to E.M.
```

Comment:
>Breaking change from C#9 which binds to E.M. [](http://example.com/codeflow?start=15&length=44)

This break needs to be considered.

Reply:
I didn't understand the comment. This seems the break we've already accepted (ie. bullet 2 in doc above, scenario where extension ends up losing)

Reply:
The argument in this case is a method group so ideally the `Expression` overload should be ignored.

Reply:
> The argument in this case is a method group so ideally the Expression overload should be ignored.

Quite possibly this is an unrelated issue, but this feels like a bug. Why the Expression candidate is considered applicable?

Reply:
Discussed in [https://github.com/dotnet/csharplang/blob/main/meetings/2021/LDM-2021-09-20.md#method-groups-converting-to-expression](https://github.com/dotnet/csharplang/blob/main/meetings/2021/LDM-2021-09-20.md#method-groups-converting-to-expression)